skbuild/cmaker.py

```diff
@@ -313,9 +314,18 @@ def configure(
         # Parse CMAKE_ARGS only if SKBUILD_CONFIGURE_OPTIONS is not present
         if "SKBUILD_CONFIGURE_OPTIONS" in os.environ:
             env_cmake_args = list(filter(None, shlex.split(os.environ["SKBUILD_CONFIGURE_OPTIONS"])))
+            if any("CMAKE_INSTALL_PREFIX" in arg for arg in env_cmake_args):
+                raise ValueError("CMAKE_INSTALL_PREFIX may not be passed via SKBUILD_CONFIGURE_OPTIONS.")
         else:
-            env_cmake_args_filtered = filter(None, shlex.split(os.environ.get("CMAKE_ARGS", "")))
+            env_cmake_args_filtered = list(filter(None, shlex.split(os.environ.get("CMAKE_ARGS", ""))))
             env_cmake_args = [s for s in env_cmake_args_filtered if "CMAKE_INSTALL_PREFIX" not in s]
+            # Don't error, only warn in this instance because CMAKE_ARGS may be out of
+            # the project's control (e.g. when provided by conda builds).
+            if len(env_cmake_args) < len(env_cmake_args_filtered):
+                warnings.warn(
+                    "CMAKE_INSTALL_PREFIX was passed via CMAKE_ARGS and ignored. This "
```

Comment:
Is it important to warn? This is _very_ common, as it will always happen on conda builds.

Reply:
No, the warning isn't really necessary. I added it for consistency since it seemed odd to have some annotation for every other possible way of providing CMake arguments, but I can remove it.

Reply:
CMAKE_ARGS is a bit special, since it's picking up something general that might not be targeted to scikit-build. The other two are always targeted at scikit-build.

Reply:
Out of curiosity, does any tool use `CMAKE_ARGS` other than conda-forge builds? What concerns me a little here is the possibility that scikit-build is being developed for compatibility with conda-forge rather than the other way around. I think it's fine for scikit-build (which aims to be a reasonably drop-in replacement for setuptools) but I want to make sure that same mindset doesn't percolate to scikit-build-core. With modern PEP 517 builders, which could all be developed completely independently of one another, I think the responsibility should be on whatever invokes a build tool (e.g. conda-forge) to set up arguments in a way that is compatible with any builder that it wants to support.